tpm2/Makefile

```diff
@@ -206,7 +206,7 @@ uninstall:
 .PHONY: clean
 clean:
 	rm -rf $(BIN_DIR)
-	rm -rf $(OBJECT_DIRS)
+	rm -rf $(OBJ_DIR)
 	rm -rf $(DOC_DIRS)
```

Comment:
It was plural to include both the logger/doc and doc directory. It looks like the intent was to generate separate documentation for the logger library, but the logger documentation was not setup. The Doxyfile only specified src/ and include/ for input directories and I think that creating the logger docs in a separate directory would have required a separate Doxyfile and an addition doxygen call in the Makefile. In any case, as the logger functionality is now in libkmyth, I'm not sure that makes sense.

For now, I changed the Doxyfile to generate documentation for anything in the current (kmyth/tpm2) directory. In addition to the logger code, this will also include the unit test code in the generated documentation. I removed the concept of a logger documentation directory from the Makefile and got rid of the additional DOC_DIRS variable used to aggregate the documentation directories. I moved the creation (mkdir -p) of the doc directory from the pre target to the doc target so that it is now only created if the documentation actually gets built.

This may or may not be the way we want to handle this longer term, but I'm thinking we can introduce an issue if we decide to do this differently.